src/components/PeopleFilters.tsx

```diff
@@ -1,12 +1,78 @@
-export const PeopleFilters = () => {
+import classNames from 'classnames';
+import { Link, useSearchParams } from 'react-router-dom';
+
+type Param = string | number;
+type Params = {
+  [key: string]: Param[] | Param | null;
+};
+
+export function getSearchWith(
+  params: Params,
+  search?: string | URLSearchParams,
+) {
+  const newParams = new URLSearchParams(search);
+  // eslint-disable-next-line
+  for (const [key, value] of Object.entries(params)) {
+    if (value === null) {
+      newParams.delete(key);
+    } else if (Array.isArray(value)) {
+      newParams.delete(key);
+      value.forEach(item => newParams.append(key, item.toString()));
+    } else {
+      newParams.set(key, value.toString());
+    }
```

Comment:
for multiple condition better to use switch case 
```suggestion
switch (true) {
  case !value:
    newParams.delete(key);
    break;
  case Array.isArray(value):
    newParams.delete(key);
    value.forEach(item => newParams.append(key, item.toString()));
    break;
  default:
    newParams.set(key, value.toString());
}

```